### PR TITLE
Replace the logging.basicConfig by a proper configuration

### PR DESCRIPTION
**Pull request by evandrocoan**
basicConfig: Do basic configuration for the logging system.
This function does nothing if the root logger already has handlers
configured. It is a convenience method intended for use by simple
scripts to do one-shot configuration of the logging package.

### CODE REVIEW

File: pyls/__main__.py [hunk 1]
```diff
@@ -95,3 +80,37 @@ def _binary_stdio():
...
+    root_logger = logging.root
+
+    if log_config:
+        with open(log_config, 'r') as f:
+            logging.config.dictConfig(json.load(f))
+    else:
+        log_file = args.log_file
+        formatter = logging.Formatter(LOG_FORMAT, style="%")
+        if log_file:
+            log_handler = logging.handlers.RotatingFileHandler(log_file, mode='a',
```
Comment (gatesn) [inline pyls/__main__.py]: Can you stick to repo parenthesis style? 
```
logging.handlers.RotatingFileHandler(
    log_file, mode='a', maxBytes=50*1024*1024, 
    backupCount=10, encoding=None, delay=0
)
```

File: pyls/__main__.py [hunk 2]
```diff
@@ -95,3 +80,37 @@ def _binary_stdio():
...
+            log_handler = logging.StreamHandler()
+        log_handler.setFormatter(formatter)
+        root_logger.addHandler(log_handler)
+
+    # CRITICAL = 50
+    # FATAL = CRITICAL
+    # ERROR = 40
+    # WARNING = 30
+    # INFO = 20
+    # DEBUG = 10
```
Comment (gatesn) [inline pyls/__main__.py]: Why are these comments here? They don't seem to be relevant to the code

File: pyls/__main__.py [hunk 3]
```diff
@@ -95,3 +80,37 @@ def _binary_stdio():
         stdin, stdout = sys.stdin, sys.stdout
 
     return stdin, stdout
+
+def configure_logger(args):
```
Comment (gatesn) [inline pyls/__main__.py]: Can you prefix function with underscore

Comment (evandrocoan) [reply pyls/__main__.py]: Sure.

File: pyls/__main__.py (hunk 1 above)
Comment (evandrocoan) [reply pyls/__main__.py]: Oops.

File: pyls/__main__.py (hunk 2 above)
Comment (evandrocoan) [reply pyls/__main__.py]: Informative about the debug levels. I put it there because I was confused about them when I saw you can only pass 3 levels, but I know the `logging` module has 6. So, looking at these values I can clearly see that we you set the WARNING level, you can also see the ERROR and other above it.

**Comment by gatesn** [comment]
Some linting errors too:

```
************* Module pyls.__main__
C: 96, 0: Wrong continued indentation (add 43 spaces).
                    maxBytes=50*1024*1024, backupCount=10, encoding=None, delay=0)
                    ^                                          | (bad-continuation)
C:116, 0: Trailing newlines (trailing-newlines)
E: 93,20: Unexpected keyword argument 'style' in constructor call (unexpected-keyword-arg)
I: 75, 0: Useless suppression of 'no-member' (useless-suppression)
I: 75, 0: Useless suppression of 'import-error' (useless-suppression)
```

**Comment by evandrocoan** [comment]
I should have run `tox` before. I am running it now.

File: pyls/__main__.py (hunk 2 above)
Comment (gatesn) [reply pyls/__main__.py]: I think that's part of the logging API and doesn't need re-documenting here. Most logging frameworks would log at the given level or "above"

**Comment by evandrocoan** [comment]
@gatesn, this linter is insane:

1. First it complained about missing 2 trailing new lines. 
1. Now it complains about a existent trailing new line.
